app/presenters/uniprot_mapping_tsv_presenter.rb

```diff
@@ -0,0 +1,34 @@
+class UniprotMappingTsvPresenter
+    def self.objects
+      Gene.joins(variants: :evidence_items).where("variants.evidence_items.status!='rejected'").distinct
+    end
+  
+    def self.headers
+      [
+        'civic_name',
+        'uniprot_name',
+        'gene_overview'
+      ]
+    end 
+  
+    def self.row_from_object(gene)
+      [
+        gene.name,
+        Scrapers::MyGeneInfo.get_swissprot_name(gene),
+        formatted_overview_col(gene)
+      ]
+    end
+  
+    def self.file_name
+      "UniprotMapping.tsv"
+    end 
+  
+    def self.formatted_overview_col(gene)
```

Comment:
I'd get the counts in the following ways:

```
eid_count = EvidenceItem.joins(variant: [:gene]).where("evidence_items.status != 'rejected'").where(variant: {gene: gene}).distinct.count
variant_count = gene.variants.joins(:evidence_items).where("evidence_items.status != 'rejected'").distinct.count
assertion_count = gene.assertions.where("status != 'rejected'").distinct.count
```

You could also invert the logic and do something like this:

```
Assertion.joins(:gene).where("status != 'rejected'").where(gene: g).distinct.count
```
depending on what's more clear to you.